File: crates/ih-muse/tests/it/test_element.rs
// tests/it/test_element.rs
use super::common::TestContext;
use ih_muse::ClientType;

#[tokio::test]
async fn test_element_registration() {
    let ctx = TestContext::new(ClientType::Poet).await;
    let local_elem_id = ctx.register_test_element().await;
    let state = ctx.muse.get_state();

    let element_id = state
        .get_element_id(&local_elem_id)
        .expect("Element was not registered");

    println!(
        "Element with LocalElementId {:?} was registered with ElementId {:?}",
        local_elem_id, element_id
    );
}

File: tests/test_element.py
# tests/test_element.py

import pytest
from common import MuseTestContext


@pytest.mark.asyncio
async def test_element_registration() -> None:
    ctx = await MuseTestContext.create()
    local_elem_id = await ctx.register_test_element()

    # Since we don't have direct access to the state, we assume the element is registered
    assert local_elem_id is not None, "Failed to register element"
